src/pages/index.js

```diff
@@ -0,0 +1,382 @@
+  },
+  {
+    name: 'A very long bridge, over the forest and through the trees',
+    link: forestBridgeImage,
+  },
+  {
+    name: 'Tunnel with morning light',
+    link: tunnelImage,
+  },
+  {
+    name: 'Mountain house',
+    link: mountainHouseImage,
+  },
+]
+
+const api = new Api({
+  baseUrl: 'https://around-api.en.tripleten-services.com/v1',
+  headers: {
+    authorization: '281d0863-c6ec-4780-a7d9-4358227ce389',
+    'Content-Type': 'application/json',
+  },
+})
+
+// Modal elements
+const editProfileBtn = document.querySelector('.profile__edit-btn')
+const editProfileModal = document.querySelector('#edit-profile-modal')
+const newPostModal = document.querySelector('#new-post-modal')
+const avatarModalBtn = document.querySelector('.profile__avatar-btn')
+const editProfileCloseBtn = editProfileModal.querySelector('.modal__close-btn')
+const editProfileForm = document.forms['edit-profile-form']
+const editProfileNameInput = editProfileForm.elements['profile-name']
+const editProfileDescriptionInput =
+  editProfileForm.elements['profile-description']
+
+// New post form elements
+const newPostBtn = document.querySelector('.profile__add-btn')
+const newPostCloseBtn = newPostModal.querySelector('.modal__close-btn')
+const newPostForm = document.forms['new-post-form']
+const newPostImageInput = newPostForm.elements['card-image']
+const newPostCaptionInput = newPostForm.elements['caption']
+
+// Avatar form elements
+const avatarModal = document.querySelector('#avatar-modal')
+const avatarCloseBtn = avatarModal.querySelector('.modal__close-btn')
+const avatarForm = avatarModal.querySelector('.modal__form')
+const avatarInput = avatarModal.querySelector('#profile-avatar-input')
+
+const profileNameEl = document.querySelector('.profile__name')
+const profileDescriptionEl = document.querySelector('.profile__description')
+
+// Delete modal elements
+const deleteModal = document.querySelector('#delete-modal')
+const deleteForm = document.querySelector('#delete-form')
+const deleteCancelBtn = deleteForm.querySelector('.modal__button_cancel')
+
+// Preview image popup elements
+const previewModal = document.querySelector('#preview-modal')
+const previewModalCloseBtn = previewModal.querySelector('.modal__close-btn')
+const previewImageEl = previewModal.querySelector('.modal__image')
+const previewCaptionEl = previewModal.querySelector('.modal__caption')
+
+// Card related elements
+let selectedCard, selectedCardId
+let currentUser // Store user info globally for access in handlers
+
+const cardTemplate = document
+  .querySelector('#card-template')
+  .content.querySelector('.card')
+const cardsList = document.querySelector('.cards__list')
+
+api
+  .getAppInfo()
+  .then(([cards, userInfo]) => {
+    currentUser = userInfo // Set current user info after API call
+    cards.forEach((item) => {
+      const cardEl = getCardEl(item)
+      cardsList.append(cardEl)
+    })
+
+    const avatarImg = document.querySelector('.profile__avatar')
```

Comment:
`document.querySelector('.profile__avatar')` 

It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine